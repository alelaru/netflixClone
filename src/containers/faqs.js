import faqs from "../fixtures/faqs.json"
import Acoordion from "../components/acoordeon"
import OptForm from "../components/opt-form"

const FaqsContainer = () => {
    return ( 
        <Acoordion>
            <Acoordion.Title>Frequently Asked Questions</Acoordion.Title>
            <Acoordion.Frame>
                {faqs.map((item) => (
                    <Acoordion.Item key={item.id}>
                    <Acoordion.Header>{item.header}</Acoordion.Header>
                    <Acoordion.Body>{item.body}</Acoordion.Body>
                    </Acoordion.Item>
                ))}
            </Acoordion.Frame>

            <OptForm>
                <OptForm.Input placeholder="Email Address"></OptForm.Input>
                <OptForm.Button>Try it now</OptForm.Button>
                <OptForm.Break></OptForm.Break>
                <OptForm.Text>Ready to watch? Enter your email to create or restart your membership</OptForm.Text>
            </OptForm>
        </Acoordion>
     );
}
 
export default FaqsContainer;