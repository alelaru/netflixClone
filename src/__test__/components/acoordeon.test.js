import { fireEvent, render } from "@testing-library/react"
import Acoordion from "../../components/acoordeon"
import faqs from "../../fixtures/faqs.json"


describe("<Acoordeon/>", () => {
    
    it("Renders the Acoordion with populated data", () => {
        const {container, getByText} = render(

            // I removed the OptForm we can test that later
            <Acoordion>
            <Acoordion.Title>Frequently Asked Questions</Acoordion.Title>
            {faqs.map((item) => (
                <Acoordion.Item key={item.id}>
                <Acoordion.Header>{item.header}</Acoordion.Header>
                <Acoordion.Body>{item.body}</Acoordion.Body>
                </Acoordion.Item>
            ))}
                <Acoordion.Item/>
            </Acoordion>
        );

        expect(getByText("Frequently Asked Questions")).toBeTruthy();
        expect(getByText("What is Netflix?")).toBeTruthy();
        expect(getByText("How much does Netflix cost?")).toBeTruthy();
        expect(getByText("Where can I watch?")).toBeTruthy();
        expect(getByText("How do I cancel?")).toBeTruthy();
        expect(getByText("What can I watch on Netflix?")).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    })

    it("opens and closes the Acoordion component", () => {
        const {container, getByText, queryByText, debug} = render(

            // I removed the OptForm we can test that later
            <Acoordion>
            <Acoordion.Title>Frequently Asked Questions</Acoordion.Title>
            <Acoordion.Frame>
                {faqs.map((item) => (
                    <Acoordion.Item key={item.id}>
                    <Acoordion.Header>{item.header}</Acoordion.Header>
                    <Acoordion.Body>{item.body}</Acoordion.Body>
                    </Acoordion.Item>
                ))}
            </Acoordion.Frame>

            </Acoordion>
        )

        const WhatCanIWatchOnNetflix = "Netflix has an extensive library of feature films, documentaries, TV programmes, anime, award-winning Netflix originals, and more. Watch as much as you want, any time you want."

        expect(queryByText(WhatCanIWatchOnNetflix)).toBeFalsy();
        fireEvent.click(getByText("What can I watch on Netflix?"))
        expect(queryByText(WhatCanIWatchOnNetflix)).toBeTruthy();

        fireEvent.click(getByText("What can I watch on Netflix?"))
        expect(queryByText(WhatCanIWatchOnNetflix)).toBeFalsy();
        expect(container.firstChild).toMatchSnapshot();
        
    })

})