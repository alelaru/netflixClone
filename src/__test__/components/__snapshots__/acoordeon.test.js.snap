// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Acoordeon/> Renders the Acoordion with populated data 1`] = `
<div
  class="acoordeon__Container-sc-1yriwl0-0 hwKyNs"
>
  <div
    class="acoordeon__Inner-sc-1yriwl0-2 hURoTZ"
  >
    <div
      class="acoordeon__Title-sc-1yriwl0-5 cLKZBx"
    >
      Frequently Asked Questions
    </div>
    <div
      class="acoordeon__Item-sc-1yriwl0-3 kxqYPA"
    >
      <div
        class="acoordeon__Header-sc-1yriwl0-4 cIrBZk"
      >
        What is Netflix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="acoordeon__Item-sc-1yriwl0-3 kxqYPA"
    >
      <div
        class="acoordeon__Header-sc-1yriwl0-4 cIrBZk"
      >
        How much does Netflix cost?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="acoordeon__Item-sc-1yriwl0-3 kxqYPA"
    >
      <div
        class="acoordeon__Header-sc-1yriwl0-4 cIrBZk"
      >
        Where can I watch?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="acoordeon__Item-sc-1yriwl0-3 kxqYPA"
    >
      <div
        class="acoordeon__Header-sc-1yriwl0-4 cIrBZk"
      >
        How do I cancel?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="acoordeon__Item-sc-1yriwl0-3 kxqYPA"
    >
      <div
        class="acoordeon__Header-sc-1yriwl0-4 cIrBZk"
      >
        What can I watch on Netflix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="acoordeon__Item-sc-1yriwl0-3 kxqYPA"
    />
  </div>
</div>
`;

exports[`<Acoordeon/> opens and closes the Acoordion component 1`] = `
<div
  class="acoordeon__Container-sc-1yriwl0-0 hwKyNs"
>
  <div
    class="acoordeon__Inner-sc-1yriwl0-2 hURoTZ"
  >
    <div
      class="acoordeon__Title-sc-1yriwl0-5 cLKZBx"
    >
      Frequently Asked Questions
    </div>
    <div
      class="acoordeon__Frame-sc-1yriwl0-1 gPgvaB"
    >
      <div
        class="acoordeon__Item-sc-1yriwl0-3 kxqYPA"
      >
        <div
          class="acoordeon__Header-sc-1yriwl0-4 cIrBZk"
        >
          What is Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="acoordeon__Item-sc-1yriwl0-3 kxqYPA"
      >
        <div
          class="acoordeon__Header-sc-1yriwl0-4 cIrBZk"
        >
          How much does Netflix cost?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="acoordeon__Item-sc-1yriwl0-3 kxqYPA"
      >
        <div
          class="acoordeon__Header-sc-1yriwl0-4 cIrBZk"
        >
          Where can I watch?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="acoordeon__Item-sc-1yriwl0-3 kxqYPA"
      >
        <div
          class="acoordeon__Header-sc-1yriwl0-4 cIrBZk"
        >
          How do I cancel?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="acoordeon__Item-sc-1yriwl0-3 kxqYPA"
      >
        <div
          class="acoordeon__Header-sc-1yriwl0-4 cIrBZk"
        >
          What can I watch on Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
